core/src/main/java/org/jruby/RubyFileTest.java

```diff
@@ -444,7 +444,7 @@ public static RubyBoolean writable_p(IRubyObject recv, IRubyObject filename) {
             return RubyFileTest.writable_p(recv, filename);
         }
 
-        @JRubyMethod(name = "zero?", required = 1)
+        @JRubyMethod(name = {"empty?", "zero?"}, required = 1)
         public static RubyBoolean zero_p(ThreadContext context, IRubyObject recv, IRubyObject filename) {
             return RubyFileTest.zero_p(context, recv, filename);
```

Comment:
@enebo I pushed a new commit that uses direct calls to `RubyFileTest` and `RubyDir` instead of `callMethod`. Is that what you where looking for? I didn't think I needed to "add a definition to RubyDir" because there is already one there for `empty?` (the method `empty_p`).

Also, by dynamic dispatch, are you referring to my use of `callMethod`? That's what I was thinking you were referring to.

Thanks in advance for your feedback. It has been helpful so far.